test/cri-containerd/container_update_test.go

```diff
@@ -0,0 +1,364 @@
+			containerImage:   imageWindowsNanoserver,
+			cmd:              []string{"cmd", "/c", "ping", "-t", "127.0.0.1"},
+		},
+		{
+			name:             "WCOW_Hypervisor",
+			requiredFeatures: []string{featureWCOWHypervisor},
+			runtimeHandler:   wcowHypervisorRuntimeHandler,
+			sandboxImage:     imageWindowsNanoserver,
+			containerImage:   imageWindowsNanoserver,
+			cmd:              []string{"cmd", "/c", "ping", "-t", "127.0.0.1"},
+		},
+		{
+			name:             "LCOW",
+			requiredFeatures: []string{featureLCOW},
+			runtimeHandler:   lcowRuntimeHandler,
+			sandboxImage:     imageLcowK8sPause,
+			containerImage:   imageLcowAlpine,
+			cmd:              []string{"top"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			requireFeatures(t, test.requiredFeatures...)
+
+			if test.runtimeHandler == lcowRuntimeHandler {
+				pullRequiredLcowImages(t, []string{test.sandboxImage})
+			} else if test.runtimeHandler == wcowHypervisorRuntimeHandler {
+				pullRequiredImages(t, []string{test.sandboxImage})
+			}
+
+			podRequest := &runtime.RunPodSandboxRequest{
+				Config: &runtime.PodSandboxConfig{
+					Metadata: &runtime.PodSandboxMetadata{
+						Name:      t.Name(),
+						Namespace: testNamespace,
+					},
+				},
+				RuntimeHandler: test.runtimeHandler,
+			}
+
+			client := newTestRuntimeClient(t)
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			podID := runPodSandbox(t, client, ctx, podRequest)
+			defer removePodSandbox(t, client, ctx, podID)
+			defer stopPodSandbox(t, client, ctx, podID)
+
+			containerRequest := &runtime.CreateContainerRequest{
+				Config: &runtime.ContainerConfig{
+					Metadata: &runtime.ContainerMetadata{
+						Name: t.Name() + "-Container",
+					},
+					Image: &runtime.ImageSpec{
+						Image: test.containerImage,
+					},
+					Command: test.cmd,
+				},
+				PodSandboxId:  podID,
+				SandboxConfig: podRequest.Config,
+			}
+
+			containerID := createContainer(t, client, ctx, containerRequest)
+			defer removeContainer(t, client, ctx, containerID)
+
+			startContainer(t, client, ctx, containerID)
+			defer stopContainer(t, client, ctx, containerID)
+
+			// make request to increase cpu shares
+			updateReq := &runtime.UpdateContainerResourcesRequest{
+				ContainerId: containerID,
+			}
+
+			var expected uint32 = 5000
+			if test.runtimeHandler == lcowRuntimeHandler {
+				updateReq.Linux = &runtime.LinuxContainerResources{
+					CpuShares: int64(expected),
+				}
+			} else {
+				updateReq.Windows = &runtime.WindowsContainerResources{
+					CpuShares: int64(expected),
+				}
+			}
+
+			if _, err := client.UpdateContainerResources(ctx, updateReq); err != nil {
+				t.Fatalf("updating container resources for %s with %v", containerID, err)
+			}
+
+			if test.runtimeHandler == lcowRuntimeHandler {
+				checkLCOWResourceLimit(t, ctx, client, containerID, "/sys/fs/cgroup/cpu/cpu.shares", uint64(expected))
```

Comment:
Usually the cgroup limits for containers are at path `/sys/fs/cgroup/cpu/containers/<limitfile>`, `/sys/fs/cgroup/memory/containers/<limitfile>` right? Do these files have the same values because we have only 1 container?

Reply:
I think that's from a UVM perspective, but this check is from within the container itself.

Reply:
Oh yeah. That makes sense.